src/widgets/table/table.rs

```diff
@@ -1197,6 +1201,37 @@ mod tests {
             ]);
             assert_eq!(buf, expected);
         }
+
+
+        /// Note that this includes a regression test for a bug where the table would not render the
+        /// correct rows when there is no selection.
+        /// <https://github.com/ratatui-org/ratatui/issues/1179>
+        #[rstest]
+        #[case::no_selction(None, ["50", "51", "52", "53", "54"], 50)]
+        #[case::selection_before_offset(20, ["20", "21", "22", "23", "24"], 20)]
+        #[case::selection_immediately_before_offset(49, ["49", "50", "51", "52", "53"], 49)]
+        #[case::selection_at_start_of_offset(50, ["50", "51", "52", "53", "54"], 50)]
+        #[case::selection_at_end_of_offset(54, ["50", "51", "52", "53", "54"], 50)]
+        #[case::selection_immediately_after_offset(55, ["51", "52", "53", "54", "55"], 51)]
+        #[case::selection_after_offset(80, ["76", "77", "78", "79", "80"], 76)]
+        fn render_with_selection_and_offset<T: Into<Option<usize>>>(
+            #[case] selected_row: T,
+            #[case] expected_items: [&str; 5],
+            #[case] expected_offset: usize,
```

Comment:
Minor nitpick: I would switch items and offset, so the simple single number is before all the items. Gets easier to read then. First this, then that, the rest is the array.

Reply:
Fair call (Noting that the order of assertions below is intentionally the opposite of this, as the buffer assertion would have more info in it than the offset.)

Reply:
6eb01268e